BUILD

```diff
@@ -143,3 +105,70 @@ py_binary(
         requirement("yarl"),
     ],
 )
+
+#pkg_files(
+#    name = "python_files",
+#    srcs = ["@python39_x86_64-unknown-linux-gnu//:files"],
+#    strip_prefix = "",
+#)
+
+#pkg_tar(
+#    name = "python_layer_tar",
+#    srcs = [":python_files"],
+#)
+
+#container_layer(
+#    name = "python_layer",
+#    directory = "/usr/local",
+#    tars = [":python_layer_tar"],
+#)
+
+#container_image(
+#    name = "python_base",
+#    base = "@amazon2_linux//image",
+#    layers = [":python_layer"],
+#)
+
+container_image(
+    name = "python_compat",
+    base = "@python39_image//image",
+    symlinks = {
+        "/usr/bin/python": "/usr/local/bin/python"
+    }
+)
+
+py_layer(
+    name = "pip_tools_deps",
+    deps = [
+        "@pypi__pip_tools//:lib",
+        "@pypi__build//:lib",
+        "@pypi__pep517//:lib",
+        "@pypi__tomli//:lib",
+        "@pypi__importlib_metadata//:lib",
+        "@pypi__colorama//:lib",
+        "@pypi__click//:lib",
+        "@pypi__pip//:lib",
+        "@pypi__setuptools//:lib",
+        "@pypi__wheel//:lib"
+    ]
+)
+
+py3_image(
+    name = "compile_linux_deps",
+    srcs = ["compile_deps.py"],
+    data = ["requirements.in"],
```

Comment:
n.b. this invalidates the image when the requirements input changes, but was the easiest way to pass the requirements into the container.

Because the deps are separated into layers it shouldn't waste too much spaces.